lib/vector.js

```diff
@@ -107,11 +112,22 @@ exports.rectangle = function rectangle(x, y, width, height, options = {}) {
         const pathOptions = this._getPathOptions(options);
         pathOptions.type = 'stroke';
         pathOptions.color = pathOptions.stroke || pathOptions.color;
+
+        this.pauseContext();
+        const xObject = new xObjectForm(this.writer, width, height);
+        xObject.getContentContext()
+            .q()
+            .d(pathOptions.dash, pathOptions.dashPhase)
+            .drawRectangle(0, 0, width, height, pathOptions)
+            .Q();
+        xObject.end();
+        this.resumeContext();
+
         const context = this.pageContext;
         context.q();
-        context.d(pathOptions.dash, pathOptions.dashPhase);
-        context.drawRectangle(nx, ny, width, height, pathOptions);
-        context.Q();
+        this._setRotationContext(context, nx, ny, options);
```

Comment:
This part can be integrated with the context class later.

Reply:
I am not sure what you are referring to about "integrating with context class later", But am glad that the pull request is finally integrated. When does the NPM package get rebuilt with these changes, or is that an automatic occurrence after the PR has been absorbed?

Thanks!